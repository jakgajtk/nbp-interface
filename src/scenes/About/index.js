import React from 'react';

class About extends Component {
  render () {
    return (
      <div>About view</div>
    )
  }
}

export default About
