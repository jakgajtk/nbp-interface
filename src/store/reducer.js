import { reducer as bees } from 'redux-bees'
import { combineReducers } from 'redux'

import CurrencyDetailReducer from '../scenes/CurrencyDetail/reducer'
import CurrencyListReducer from '../scenes/CurrencyList/reducer'
import FavouriteReducer from '../scenes/Favourite/reducer'

export default combineReducers({
  bees,
  currencyDetail: CurrencyDetailReducer,
  currencyList: CurrencyListReducer,
  favourite: FavouriteReducer
})
