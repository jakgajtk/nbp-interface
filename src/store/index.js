import { createStore, applyMiddleware } from 'redux'
import { middleware as beesMiddleware } from 'redux-bees'
import { createLogger } from 'redux-logger'

import reducer from './reducer'

const store = createStore(reducer, applyMiddleware(createLogger(), beesMiddleware()))

export default store
