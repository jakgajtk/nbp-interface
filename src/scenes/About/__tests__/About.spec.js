import React from 'react'
import About from '../../About'
import renderer from 'react-test-renderer'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import { Provider } from 'react-redux'
import store from '../../../store'

it('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><MuiThemeProvider><About /></MuiThemeProvider></Provider>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})