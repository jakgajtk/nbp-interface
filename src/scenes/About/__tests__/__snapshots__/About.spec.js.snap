// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "boxSizing": "border-box",
          "cursor": "pointer",
          "fontWeight": 500,
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "paddingRight": "90px",
            "verticalAlign": "top",
            "whiteSpace": "normal",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 15,
              "muiPrepared": true,
            }
          }
        >
          Implementation of Web User Interface using NBP API.
        </span>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }
        />
      </div>
      <button
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "border": 10,
            "bottom": 0,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 0,
            "fontWeight": "inherit",
            "height": 48,
            "margin": "auto",
            "muiPrepared": true,
            "outline": "none",
            "overflow": "visible",
            "padding": 12,
            "position": "absolute",
            "right": 4,
            "textDecoration": "none",
            "top": 0,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": 48,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fill": "currentColor",
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</div>
`;
